models/classes/user/events/dispatcher/LtiUserEventDispatcher.php

```diff
@@ -0,0 +1,39 @@
+<?php
+
+/**
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public License
+ * as published by the Free Software Foundation; under version 2
+ * of the License (non-upgradable).
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ *
+ * Copyright (c) 2020 (original work) Open Assessment Technologies SA;
+ */
+
+namespace oat\taoLti\models\classes\user\events\dispatcher;
+
+use oat\oatbox\event\EventManagerAwareTrait;
+use oat\oatbox\service\ConfigurableService;
+use oat\taoLti\models\classes\LtiRoles;
+use oat\taoLti\models\classes\user\events\LtiTestTakerCreatedEvent;
+use oat\taoLti\models\classes\user\LtiUserInterface;
+
+class LtiUserEventDispatcher extends ConfigurableService
+{
+    use EventManagerAwareTrait;
+
+    public function dispatch(LtiUserInterface $ltiUser): void
+    {
+        if (in_array(LtiRoles::CONTEXT_LEARNER, $ltiUser->getRoles())) {
```

Comment:
let's ask confirmation about the role ,  just to be safe

Reply:
agreed on the role

Reply:
I think this is the only role that matches: https://github.com/oat-sa/extension-tao-lti/blob/master/models/classes/LtiRoles.php#L37

@siwane Can you help with confirmation?